tests/nested/manual/passphrase-support-on-hybrid/task.yaml

```diff
@@ -0,0 +1,196 @@
+  echo 'console=ttyS0 systemd.journald.forward_to_console=1' > pc-gadget/cmdline.extra
+  echo "Sign the shim binary"
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+  tests.nested secboot-sign gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack --filename=pc.snap pc-gadget/
+
+  # Retrieve kernel.
+  snap download --basename=pc-kernel --channel="$KERNEL_CHANNEL" pc-kernel
+  # Build kernel with initramfs with the compiled snap-bootstrap
+  uc24_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$NESTED_ASSETS_DIR"
+  mv "${NESTED_ASSETS_DIR}"/pc-kernel_*.snap pc-kernel.snap
+
+  # Create new disk for the installer to work on and attach to VM.
+  truncate --size=6G disk.img
+
+  # setup_nested_hybrid_system.sh runs the muinstaller to install a hybrid
+  # system.
+  gendeveloper1 sign-model < "$TESTSLIB"/assertions/developer1-2410-classic-dangerous.json > classic.model
+  "${TESTSTOOLS}"/setup_nested_hybrid_system.sh \
+     --model classic.model \
+     --store-dir "${STORE_DIR}" \
+     --gadget pc.snap \
+     --gadget-assertion pc.assert \
+     --kernel pc-kernel.snap \
+     --kernel-assertion pc-kernel.assert \
+     --passphrase "$PASSPHRASE" \
+     --extra-muinstaller-args "-kdf-type $KDF_TYPE" \
+     --disk disk.img
+
+  # Basic things look fine.
+  remote.exec "cat /etc/os-release" | MATCH 'NAME="Ubuntu"'
+  remote.exec "snap changes" | MATCH "Done.* Initialize system state"
+  remote.exec "snap list" | MATCH pc-kernel
+
+  # Check encryption.
+  remote.exec "sudo test -d /var/lib/snapd/device/fde"
+  remote.exec "sudo test -e /var/lib/snapd/device/fde/marker"
+  remote.exec "sudo test -e /var/lib/snapd/device/fde/marker"
+  remote.exec "sudo blkid /dev/disk/by-label/ubuntu-data-enc" | MATCH crypto_LUKS
+
+  # Check disk mappings.
+  # TODO: no ubuntu-save right now because:
+  #       "ERROR cannot store device key pair: internal error: cannot access device keypair manager if ubuntu-save is unavailable"
+  #DISK_MAPPINGS=(/run/mnt/ubuntu-save/device/disk-mapping.json
+  #               /run/mnt/data/var/lib/snapd/device/disk-mapping.json)
+  DISK_MAPPINGS=(/run/mnt/data/var/lib/snapd/device/disk-mapping.json)
+  for DM in "${DISK_MAPPINGS[@]}"; do
+      remote.exec "sudo cat $DM" > mapping.json
+      gojq -r '.pc."structure-encryption"."ubuntu-save".method' < mapping.json | MATCH LUKS 
+      gojq -r '.pc."structure-encryption"."ubuntu-data".method' < mapping.json | MATCH LUKS
+  done
+
+  # Check that on an already provisioned system the API will give a
+  # sensible reason why the system cannot be installed without further
+  # action.
+  remote.exec "sudo snap debug api /v2/systems/classic" > system
+  gojq '.result."storage-encryption".support' < system | MATCH "unavailable"
+  gojq '.result."storage-encryption"."unavailable-reason"' < system | MATCH "not encrypting device storage as checking TPM gave: the TPM is in DA lockout mode"
```

Comment:
This is a bit confusing. I am not sure why we test this here.

Reply:
I copied that check from muinstaller-real. you are right, it is not relevant here.